src/LanguageServer/Protocol/Handler/Extensions/ExtensionException.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Extensions;
+
+internal sealed record class ExtensionException(
+    [property: JsonPropertyName("typeName")] string TypeName,
```

Comment:
I'm fine with this - however I would have naively started with serializing System.Exception - I assume we don't want that here?  I know the exception type may not exist on the other side, but that is something jsonrpc already has to deal with when we throw exceptions for example.